<!doctype html><html lang=en-us><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=generator content="Wowchemy 5.5.0 for Hugo"><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=preload as=style href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap"><link rel=stylesheet href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap" media=print onload='this.media="all"'><meta name=author content="Dr. Thorsten Beier"><meta name=description content="A highly-customizable Hugo academic resume theme powered by Wowchemy website builder."><link rel=alternate hreflang=en-us href=https://derthorsten.github.io/page/><meta name=theme-color content="#1565c0"><link rel=stylesheet href=/page/css/vendor-bundle.min.c7b8d9abd591ba2253ea42747e3ac3f5.css media=print onload='this.media="all"'><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css integrity="sha512-W0xM4mr6dEP9nREo7Z9z+9X70wytKvMGeDsj7ps2+xg5QPrEBXC8tAW1IFnzjR6eoJ90JmCnFzerQJTLzIEHjA==" crossorigin=anonymous media=print onload='this.media="all"'><link rel=stylesheet href=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/styles/github.min.css crossorigin=anonymous title=hl-light media=print onload='this.media="all"'><link rel=stylesheet href=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/styles/dracula.min.css crossorigin=anonymous title=hl-dark media=print onload='this.media="all"' disabled><link rel=stylesheet href=/page/css/wowchemy.1052fab8b7700a3dc49ee23683097d66.css><script src=https://identity.netlify.com/v1/netlify-identity-widget.js></script>
<link rel=alternate href=/page/index.xml type=application/rss+xml title="Dr. Thorsten Beier"><link rel=manifest href=/page/manifest.webmanifest><link rel=icon type=image/png href=/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_32x32_fill_lanczos_center_3.png><link rel=apple-touch-icon type=image/png href=/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_180x180_fill_lanczos_center_3.png><link rel=canonical href=https://derthorsten.github.io/page/><meta property="twitter:card" content="summary"><meta property="twitter:site" content="@wowchemy"><meta property="twitter:creator" content="@wowchemy"><meta property="og:site_name" content="Dr. Thorsten Beier"><meta property="og:url" content="https://derthorsten.github.io/page/"><meta property="og:title" content="Dr. Thorsten Beier"><meta property="og:description" content="A highly-customizable Hugo academic resume theme powered by Wowchemy website builder."><meta property="og:image" content="https://derthorsten.github.io/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_512x512_fill_lanczos_center_3.png"><meta property="twitter:image" content="https://derthorsten.github.io/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_512x512_fill_lanczos_center_3.png"><meta property="og:locale" content="en-us"><meta property="og:updated_time" content="2022-08-29T11:00:00+00:00"><script type=application/ld+json>{"@context":"https://schema.org","@type":"WebSite","potentialAction":{"@type":"SearchAction","target":"https://derthorsten.github.io/page?q={search_term_string}","query-input":"required name=search_term_string"},"url":"https://derthorsten.github.io/page"}</script><title>Dr. Thorsten Beier</title></head><body id=top data-spy=scroll data-offset=70 data-target=#navbar-main class=page-wrapper><script src=/page/js/wowchemy-init.min.2ed908358299dd7ab553faae685c746c.js></script><aside class=search-modal id=search><div class=container><section class=search-header><div class="row no-gutters justify-content-between mb-3"><div class=col-6><h1>Search</h1></div><div class="col-6 col-search-close"><a class=js-search href=# aria-label=Close><i class="fas fa-times-circle text-muted" aria-hidden=true></i></a></div></div><div id=search-box><input name=q id=search-query placeholder=Search... autocapitalize=off autocomplete=off autocorrect=off spellcheck=false type=search class=form-control aria-label=Search...></div></section><section class=section-search-results><div id=search-hits></div></section></div></aside><div class=page-header><header class=header--fixed><nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main><div class=container-xl><div class="d-none d-lg-inline-flex"><a class=navbar-brand href=/>Dr. Thorsten Beier</a></div><button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar-content aria-expanded=false aria-label="Toggle navigation">
<span><i class="fas fa-bars"></i></span></button><div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none"><a class=navbar-brand href=/>Dr. Thorsten Beier</a></div><div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content><ul class="navbar-nav d-md-inline-flex"><li class=nav-item><a class=nav-link href=/page/#about><span>Home</span></a></li><li class=nav-item><a class=nav-link href=/page/#posts><span>Posts</span></a></li><li class=nav-item><a class=nav-link href=/page/#projects><span>Projects</span></a></li><li class=nav-item><a class=nav-link href=/page/#talks><span>Talks</span></a></li><li class=nav-item><a class=nav-link href=/page/#featured><span>Publications</span></a></li><li class=nav-item><a class=nav-link href=/page/#contact><span>Contact</span></a></li></ul></div><ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2"><li class=nav-item><a class="nav-link js-search" href=# aria-label=Search><i class="fas fa-search" aria-hidden=true></i></a></li><li class="nav-item dropdown theme-dropdown"><a href=# class=nav-link data-toggle=dropdown aria-haspopup=true aria-label="Display preferences"><i class="fas fa-moon" aria-hidden=true></i></a><div class=dropdown-menu><a href=# class="dropdown-item js-set-theme-light"><span>Light</span></a>
<a href=# class="dropdown-item js-set-theme-dark"><span>Dark</span></a>
<a href=# class="dropdown-item js-set-theme-auto"><span>Automatic</span></a></div></li></ul></div></nav></header></div><div class=page-body><span class="js-widget-page d-none"></span><section id=about class="home-section wg-about"><div class=home-section-bg></div><div class=container><div class=row><div class="col-12 col-lg-4"><div id=profile><img class="avatar avatar-circle" width=270 height=270 src=/page/authors/admin/avatar_hu52a603635ecebd45650b162dadabb4e5_12861_270x270_fill_q75_lanczos_center.jpg alt="Dr. Thorsten Beier"><div class=portrait-title><h2>Dr. Thorsten Beier</h2><h3>Scientific programmer</h3><h3><a href=https://www.quantstack.net target=_blank rel=noopener><span>QuantStack</span></a></h3></div><ul class=network-icon aria-hidden=true><li><a href=/#contact aria-label=envelope><i class="fas fa-envelope big-icon"></i></a></li><li><a href=https://twitter.com/ThorstenBeier target=_blank rel=noopener aria-label=twitter><i class="fab fa-twitter big-icon"></i></a></li><li><a href="https://scholar.google.de/citations?user=zoHhaQ4AAAAJ&hl=de" target=_blank rel=noopener aria-label=graduation-cap><i class="fas fa-graduation-cap big-icon"></i></a></li><li><a href=https://github.com/DerThorsten target=_blank rel=noopener aria-label=github><i class="fab fa-github big-icon"></i></a></li><li><a href=https://www.linkedin.com/in/thorsten-beier-95b121189/ target=_blank rel=noopener aria-label=linkedin><i class="fab fa-linkedin big-icon"></i></a></li></ul></div></div><div class="col-12 col-lg-8"><h1>Biography</h1><div class=article-style><p>I am Thorsten Beier, a Scientific Software Developer at QuantStack. I got his PhD at the University of Heidelberg (Germany) on Multicut Algorithms for Neurite Segmentation.
I contributed to various open source projects for image processing and discrete optimization. I am coauthor of OpenGM2, a library for inference in discrete graphical models, author of Nifty, a library for graph based image processing and contributor to VIGRA, a generic image processing library.</p></div><div class=row><div class=col-md-5><div class=section-subheading>Interests</div><ul class="ul-interests mb-0"><li>Free Open Source Software</li><li>Artificial Intelligence / Deep Learning</li><li>Image Processing</li><li>Discrete Optimization / Polyhedral Optimization</li></ul></div><div class=col-md-7><div class=section-subheading>Education</div><ul class="ul-edu fa-ul mb-0"><li><i class="fa-li fas fa-graduation-cap"></i><div class=description><p class=course>PhD in ComputerScience, 2018</p><p class=institution>University of Heidelberg</p></div></li><li><i class="fa-li fas fa-graduation-cap"></i><div class=description><p class=course>Master in ComputerScience, 2014</p><p class=institution>University of Heidelberg</p></div></li><li><i class="fa-li fas fa-graduation-cap"></i><div class=description><p class=course>BSc in Physics, 2012</p><p class=institution>University of Heidelberg</p></div></li></ul></div></div></div></div></div></section><section id=skills class="home-section wg-featurette"><div class=home-section-bg></div><div class=container><div class="row featurette"><div class="col-md-12 section-heading"><h1>Skills</h1></div><div class="col-12 col-sm-4"><div class=featurette-icon><img src=/page/media/icons/cpp.svg alt=cpp class="svg-icon svg-baseline" loading=lazy></div><div class=section-subheading>C++</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><i class="fab fa-python"></i></div><div class=section-subheading>Python</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><img src=/page/media/icons/latex.svg alt=latex class="svg-icon svg-baseline" loading=lazy></div><div class=section-subheading>LaTex</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><img src=/page/media/icons/wasm.svg alt=wasm class="svg-icon svg-baseline" loading=lazy></div><div class=section-subheading>Wasm</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><img src=/page/media/icons/conda_logo.svg alt=conda_logo class="svg-icon svg-baseline" loading=lazy></div><div class=section-subheading>conda</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><img src=/page/media/icons/jupyter.svg alt=jupyter class="svg-icon svg-baseline" loading=lazy></div><div class=section-subheading>Jupyter</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><i class="fas fa-chart-line"></i></div><div class=section-subheading>Statistics</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><i class="fas fa-calculator"></i></div><div class=section-subheading>Discrete Optimization</div></div><div class="col-12 col-sm-4"><div class=featurette-icon><i class="fas fa-atom"></i></div><div class=section-subheading>Physics</div></div></div></div></section><section id=experience class="home-section wg-experience"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Experience</h1></div><div class="col-12 col-lg-8"><div class="row experience"><div class="col-auto text-center flex-column d-none d-sm-flex"><div class="row h-50"><div class=col>&nbsp;</div><div class=col>&nbsp;</div></div><div class=m-2><span class="badge badge-pill border exp-fill">&nbsp;</span></div><div class="row h-50"><div class="col border-right">&nbsp;</div><div class=col>&nbsp;</div></div></div><div class="col py-2"><div class=card><div class=card-body><div class="d-flex align-content-start"><div class="mr-2 mb-2"><a href=https://www.quantstack.net target=_blank rel=noopener><img src=/page/media/icons/brands/quantstack.svg width=56px height=56px alt=QuantStack loading=lazy></a></div><div><div class="section-subheading card-title exp-title text-muted my-0">Scientific Programmer</div><div class="section-subheading card-title exp-company text-muted my-0"><a href=https://www.quantstack.net target=_blank rel=noopener>QuantStack</a></div><div class="text-muted exp-meta">Jan 2021 –
Present
<span class=middot-divider></span>
<span>Paris/Berlin</span></div></div></div><div class=card-text><p>Responsibilities include:</p><ul><li>Jupyter</li><li>JupyterLite</li><li>Xeus</li><li>Xtensor</li></ul></div></div></div></div></div><div class="row experience"><div class="col-auto text-center flex-column d-none d-sm-flex"><div class="row h-50"><div class="col border-right">&nbsp;</div><div class=col>&nbsp;</div></div><div class=m-2><span class="badge badge-pill border">&nbsp;</span></div><div class="row h-50"><div class="col border-right">&nbsp;</div><div class=col>&nbsp;</div></div></div><div class="col py-2"><div class=card><div class=card-body><div class="d-flex align-content-start"><div class="mr-2 mb-2"><a href=https://www.quality-match.com/ target=_blank rel=noopener><img src=/page/media/icons/brands/qm.svg width=56px height=56px alt="Quality Match" loading=lazy></a></div><div><div class="section-subheading card-title exp-title text-muted my-0">Backend Developer</div><div class="section-subheading card-title exp-company text-muted my-0"><a href=https://www.quality-match.com/ target=_blank rel=noopener>Quality Match</a></div><div class="text-muted exp-meta">Jan 2020 –
Jan 2021
<span class=middot-divider></span>
<span>Heidelberg</span></div></div></div><div class=card-text>Implemented a backend for labeling / crowd-sourcing tool.</div></div></div></div></div><div class="row experience"><div class="col-auto text-center flex-column d-none d-sm-flex"><div class="row h-50"><div class="col border-right">&nbsp;</div><div class=col>&nbsp;</div></div><div class=m-2><span class="badge badge-pill border">&nbsp;</span></div><div class="row h-50"><div class="col border-right">&nbsp;</div><div class=col>&nbsp;</div></div></div><div class="col py-2"><div class=card><div class=card-body><div class="d-flex align-content-start"><div class="mr-2 mb-2"><a href=https://www.embl.org/sites/heidelberg/ target=_blank rel=noopener><img src=/page/media/icons/brands/embl.svg width=56px height=56px alt=EMBL loading=lazy></a></div><div><div class="section-subheading card-title exp-title text-muted my-0">Bioinformatician</div><div class="section-subheading card-title exp-company text-muted my-0"><a href=https://www.embl.org/sites/heidelberg/ target=_blank rel=noopener>EMBL</a></div><div class="text-muted exp-meta">Jan 2019 –
Jan 2020
<span class=middot-divider></span>
<span>Heidelberg</span></div></div></div><div class=card-text>Maintained <a href=https://kipoi.org/ target=_blank rel=noopener>https://kipoi.org/</a></div></div></div></div></div><div class="row experience"><div class="col-auto text-center flex-column d-none d-sm-flex"><div class="row h-50"><div class="col border-right">&nbsp;</div><div class=col>&nbsp;</div></div><div class=m-2><span class="badge badge-pill border">&nbsp;</span></div><div class="row h-50"><div class=col>&nbsp;</div><div class=col>&nbsp;</div></div></div><div class="col py-2"><div class=card><div class=card-body><div class="d-flex align-content-start"><div class="mr-2 mb-2"><a href=https://www.heidelbergengineering.com target=_blank rel=noopener><img src=/page/media/icons/brands/heidelberg_engineering.svg width=56px height=56px alt="Heidelberg Engineering" loading=lazy></a></div><div><div class="section-subheading card-title exp-title text-muted my-0">Digital Image Processing Specialist</div><div class="section-subheading card-title exp-company text-muted my-0"><a href=https://www.heidelbergengineering.com target=_blank rel=noopener>Heidelberg Engineering</a></div><div class="text-muted exp-meta">Jan 2018 –
Jan 2019
<span class=middot-divider></span>
<span>Heidelberg</span></div></div></div><div class=card-text>Implemented algorithms for ophthalmology.</div></div></div></div></div></div></div></div></section><section id=posts class="home-section wg-pages"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Recent Posts</h1></div><div class="col-12 col-lg-8"><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/post/mamba-meets-jupyterlite/>Mamba meets JupyterLite</a></div><a href=/page/post/mamba-meets-jupyterlite/ class=summary-link><div class=article-style>Introducing a mamba-based distribution for WebAssembly, and deploying scalable computing environments with JupyterLite.</div></a><div class="stream-meta article-metadata"><div class=article-metadata><div><span class=author-highlighted>Dr. Thorsten Beier</span></div><span class=article-date>Jul 14, 2022</span>
<span class=middot-divider></span>
<span class=article-reading-time>1 min read</span>
<span class=middot-divider></span>
<span class=article-categories><i class="fas fa-folder mr-1"></i><a href=/page/category/jupyterlite/>JupyterLite</a>, <a href=/page/category/emscripten-forge/>emscripten-forge</a></span></div></div></div><div class=ml-3><a href=/page/post/mamba-meets-jupyterlite/><img src=/page/post/mamba-meets-jupyterlite/featured_huc656b89d92bcf4e690a04d77bc52b489_88744_150x0_resize_q75_h2_lanczos_3.webp height=47 width=150 alt="Mamba meets JupyterLite" loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/post/xeus-lite/>Xeus-Lite</a></div><a href=/page/post/xeus-lite/ class=summary-link><div class=article-style>JupyterLite is a JupyterLab distribution that runs entirely in the web browser, backed by in-browser language kernels. Xeus is C++ library for writing Jupyter kernels. In this blogpost we show how these two can work together.</div></a><div class="stream-meta article-metadata"><div class=article-metadata><div><span class=author-highlighted>Dr. Thorsten Beier</span></div><span class=article-date>Jul 14, 2022</span>
<span class=middot-divider></span>
<span class=article-reading-time>1 min read</span>
<span class=middot-divider></span>
<span class=article-categories><i class="fas fa-folder mr-1"></i><a href=/page/category/jupyterlite/>JupyterLite</a>, <a href=/page/category/xeus/>Xeus</a></span></div></div></div><div class=ml-3><a href=/page/post/xeus-lite/><img src=/page/post/xeus-lite/featured_hu0ce0f76fe6e4bd1087333bb798551abf_383931_150x0_resize_q75_h2_lanczos_3.webp height=118 width=150 alt=Xeus-Lite loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/post/jupyter-games/>Jupyter-Games</a></div><a href=/page/post/jupyter-games/ class=summary-link><div class=article-style>Making their own tiny video games can be a great way for kids to learn programming in a playful matter. While Jupyter is widely used as a scientific and educational tool, Jupyter is seldom used as a platform for game development. In this blog post, we show how Jupyter can be used to develop tiny games based on Box2D. While Jupyter is language agnostic and kernels exist for many programming languages, we will focus on the Python programming language in this blog post.</div></a><div class="stream-meta article-metadata"><div class=article-metadata><div><span class=author-highlighted>Dr. Thorsten Beier</span></div><span class=article-date>Dec 14, 2021</span>
<span class=middot-divider></span>
<span class=article-reading-time>1 min read</span>
<span class=middot-divider></span>
<span class=article-categories><i class="fas fa-folder mr-1"></i><a href=/page/category/jupyter/>Jupyter</a>, <a href=/page/category/games/>Games</a>, <a href=/page/category/box2d/>Box2D</a></span></div></div></div><div class=ml-3><a href=/page/post/jupyter-games/><img src=/page/post/jupyter-games/featured_hud0e3526097145c7d0063a2e96b5a84d0_112060_150x0_resize_q75_h2_lanczos_3.webp height=74 width=150 alt=Jupyter-Games loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/post/from-jupyter-to-the-moon/>From Jupyter to the Moon</a></div><a href=/page/post/from-jupyter-to-the-moon/ class=summary-link><div class=article-style>Introducing a new Jupyter kernel for Lua based on Xeus</div></a><div class="stream-meta article-metadata"><div class=article-metadata><div><span class=author-highlighted>Dr. Thorsten Beier</span></div><span class=article-date>Sep 24, 2021</span>
<span class=middot-divider></span>
<span class=article-reading-time>1 min read</span>
<span class=middot-divider></span>
<span class=article-categories><i class="fas fa-folder mr-1"></i><a href=/page/category/jupyter/>Jupyter</a>, <a href=/page/category/games/>Games</a>, <a href=/page/category/box2d/>Box2D</a></span></div></div></div><div class=ml-3><a href=/page/post/from-jupyter-to-the-moon/><img src=/page/post/from-jupyter-to-the-moon/featured_hua7099ef39e7e5515522953d8200c1519_76651_150x0_resize_q75_h2_lanczos_3.webp height=61 width=150 alt="From Jupyter to the Moon" loading=lazy></a></div></div></div></div></div></section><section id=projects class="home-section wg-portfolio"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Projects</h1></div><div class="col-12 col-lg-8"><div class="row js-layout-row"><div class="col-12 isotope-item js-id-emscripten-forge js-id-wasm js-id-conda-forge"><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/project/emscripten-forge/>Emscripten-Forge</a></div><a href=/page/project/emscripten-forge/ class=summary-link><div class=article-style>A conda channel for the <code>emscripten-32</code> platform</div></a><div class="stream-meta article-metadata"></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/emscripten-forge/recipes target=_blank rel=noopener><i class="fab fa-gitub mr-1"></i>code</a></div></div><div class=ml-3></div></div></div><div class="col-12 isotope-item js-id-JupyterLite js-id-Wasm js-id-Xeus"><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/project/xeus-lite/>xeus-lite</a></div><a href=/page/project/xeus-lite/ class=summary-link><div class=article-style>A wasm build of the xeus library for writing JupyterLite kernels</div></a><div class="stream-meta article-metadata"></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/pyb2d/pyb2d target=_blank rel=noopener><i class="fab fa-gitub mr-1"></i>code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href="https://jupyterlite.github.io/demo/lab?path=pyolite%2Fpyb2d%2Fnewtons_cradle.ipynb" target=_blank rel=noopener><i class="custom custom-jupyter mr-1"></i>Try in JupyterLite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href="https://mybinder.org/v2/gh/pyb2d/pyb2d/main?urlpath=/lab/tree/examples/jupyter_integration.ipynb" target=_blank rel=noopener><i class="custom custom-jupyter mr-1"></i>Try in Jupyter</a></div></div><div class=ml-3><a href=/page/project/xeus-lite/><img src=/page/project/xeus-lite/featured_hu2aacd5877c743429ecf71e3a6ff6e373_624278_150x0_resize_q75_h2_lanczos.webp height=1266 width=1316 alt=xeus-lite loading=lazy></a></div></div></div><div class="col-12 isotope-item js-id-discrete-optimization"><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/project/opengm/>OpenGM</a></div><a href=/page/project/opengm/ class=summary-link><div class=article-style>An C++ library for discrete optimization</div></a><div class="stream-meta article-metadata"></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=http://www.andres.sc/publications/opengm-2.0.2-beta-manual.pdf target=_blank rel=noopener>PDF</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/opengm/opengm target=_blank rel=noopener><i class="fab fa-gitub mr-1"></i>code</a></div></div><div class=ml-3></div></div></div><div class="col-12 isotope-item js-id-Discrete-Optimization js-id-Multicut js-id-Lifted-Multicut js-id-Agglomerative-Clustering js-id-C++"><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/project/nifty/>nifty</a></div><a href=/page/project/nifty/ class=summary-link><div class=article-style>A C++ / Python library for graph based image processing, solving the multicut and lifted multicut objective, and agglomerative clustering.</div></a><div class="stream-meta article-metadata"></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/DerThorsten/nifty target=_blank rel=noopener>Code</a></div></div><div class=ml-3></div></div></div><div class="col-12 isotope-item js-id-Deep-Learning"><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/project/pyb2d/>pyb2d</a></div><a href=/page/project/pyb2d/ class=summary-link><div class=article-style>Python bindings for the Box2D library</div></a><div class="stream-meta article-metadata"></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/pyb2d/pyb2d target=_blank rel=noopener><i class="fab fa-gitub mr-1"></i>code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href="https://jupyterlite.github.io/demo/lab?path=pyolite%2Fpyb2d%2Fnewtons_cradle.ipynb" target=_blank rel=noopener><i class="custom custom-jupyter mr-1"></i>Try in JupyterLite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href="https://mybinder.org/v2/gh/pyb2d/pyb2d/main?urlpath=/lab/tree/examples/jupyter_integration.ipynb" target=_blank rel=noopener><i class="custom custom-jupyter mr-1"></i>Try in Jupyter</a></div></div><div class=ml-3><a href=/page/project/pyb2d/><img src=/page/project/pyb2d/featured_hu2aacd5877c743429ecf71e3a6ff6e373_624278_150x0_resize_q75_h2_lanczos.webp height=1266 width=1316 alt=pyb2d loading=lazy></a></div></div></div></div></div></div></div></section><section id=talks class="home-section wg-pages"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Recent & Upcoming Talks</h1></div><div class="col-12 col-lg-8"><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/talk/euroscipy-2022-interactive-data-science-in-the-browser-with-jupyterlite-and-emscripten-forge/>EuroSciPy 2022 - Interactive Data Science in the browser with JupyterLite and Emscripten Forge</a></div><a href=/page/talk/euroscipy-2022-interactive-data-science-in-the-browser-with-jupyterlite-and-emscripten-forge/ class=summary-link><div class=article-style>JupyterLite is a JupyterLab distribution that runs entirely in the web browser, backed by in-browser language kernels including the WebAssembly powered Pyodide kernel.</div></a><div class="stream-meta article-metadata"><div><span>Aug 29, 2022 11:00 AM &mdash; Sep 2, 2022 11:00 AM</span></div><div><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Jeremy Tuloup</span>, <span>Martin RenouThorsten Beier</span></div></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/jtpio/euroscipy-2022-jupyterlite-emscripten-forge target=_blank rel=noopener>Code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href="https://euroscipy-2022-jupyterlite-emscripten-forge.readthedocs.io/en/latest/lite/lab/index.html?path=presentation.ipynb" target=_blank rel=noopener>Slides</a></div></div><div class=ml-3><a href=/page/talk/euroscipy-2022-interactive-data-science-in-the-browser-with-jupyterlite-and-emscripten-forge/><img src=/page/talk/euroscipy-2022-interactive-data-science-in-the-browser-with-jupyterlite-and-emscripten-forge/featured_hu8101cdb2578a66e1413806a5bcabe279_85303_150x0_resize_q75_h2_lanczos_3.webp height=97 width=150 alt="EuroSciPy 2022 - Interactive Data Science in the browser with JupyterLite and Emscripten Forge" loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/talk/jupyterlite-jupyter-webassembly-python/>JupyterLite: Jupyter ❤️ WebAssembly ❤️ Python</a></div><a href=/page/talk/jupyterlite-jupyter-webassembly-python/ class=summary-link><div class=article-style>JupyterLite is a JupyterLab distribution that runs entirely in the web browser, backed by in-browser language kernels including the WebAssembly powered Pyodide kernel.</div></a><div class="stream-meta article-metadata"><div><span>Oct 29, 2021 11:00 AM &mdash; 11:30 AM</span></div><div><span>Jeremy Tuloup</span>, <span>Madhur Tandon</span>, <span>Martin Renou</span>, <span class=author-highlighted>Dr. Thorsten Beier</span></div></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href="https://www.youtube.com/watch?v=EzclRzL8rt0" target=_blank rel=noopener>Video</a></div></div><div class=ml-3><a href=/page/talk/jupyterlite-jupyter-webassembly-python/><img src=/page/talk/jupyterlite-jupyter-webassembly-python/featured_hu8101cdb2578a66e1413806a5bcabe279_85303_150x0_resize_q75_h2_lanczos_3.webp height=97 width=150 alt="JupyterLite: Jupyter ❤️ WebAssembly ❤️ Python" loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/talk/an-efficient-fusion-move-algorithm-for-the-minimum-cost-lifted-multicut-problem/>An efficient fusion move algorithm for the minimum cost lifted multicut problem</a></div><a href=/page/talk/an-efficient-fusion-move-algorithm-for-the-minimum-cost-lifted-multicut-problem/ class=summary-link><div class=article-style>Many computer vision problems can be cast as an optimization problem whose feasible solutions are decompositions of a graph. The minimum cost lifted multicut problem is such an optimization problem. Its objective function can penalize or reward all decompositions for which any given pair of nodes are in distinct components. While this property has many potential applications, such applications are hampered by the fact that the problem is NP-hard. We propose a fusion move algorithm for computing feasible solutions, better and more efficiently than existing algorithms. We demonstrate this and applications to image segmentation, obtaining a new state of the art for a problem in biological image analysis.</div></a><div class="stream-meta article-metadata"><div><span>Oct 8, 2016 11:00 AM &mdash; 11:30 AM</span></div><div><span>Thorsten Beier</span>, <span>Björn Andres</span>, <span>Ullrich Köthe</span>, <span>Fred A Hamprecht</span></div></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href="http://videolectures.net/site/normal_dl/tag=1078753/eccv2016_beier_efficient_fusion_01.pdf" target=_blank rel=noopener>Slides</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=http://videolectures.net/eccv2016_beier_efficient_fusion/ target=_blank rel=noopener>Video</a></div></div><div class=ml-3><a href=/page/talk/an-efficient-fusion-move-algorithm-for-the-minimum-cost-lifted-multicut-problem/><img src=/page/talk/an-efficient-fusion-move-algorithm-for-the-minimum-cost-lifted-multicut-problem/featured_huc98ef8fcaa275e8b19742dd09046008c_262797_150x0_resize_q75_h2_lanczos_3.webp height=95 width=150 alt="An efficient fusion move algorithm for the minimum cost lifted multicut problem" loading=lazy></a></div></div></div></div></div></section><section id=featured class="home-section wg-featured"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Featured Publications</h1></div><div class="col-12 col-lg-8"><div class="card-simple view-card"><div class=article-metadata><div><span>Stuart Berg</span>, <span>Dominik Kutra</span>, <span>Thorben Kroeger</span>, <span>Christoph N. Straehle</span>, <span>Bernhard X. Kausler</span>, <span>Carsten Haubold</span>, <span>Martin Schiegg</span>, <span>Janez Ales</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Markus Rudy</span>, <span>Kemal Eren</span>, <span>Jaime I Cervantes</span>, <span>Buote Xu</span>, <span>Fynn Beuttenmueller</span>, <span>Adrian Wolny</span>, <span>Chong Zhang</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span>, <span>Anna Kreshuk</span></div><span class=article-date>December, 2019</span>
<span class=middot-divider></span>
<span class=pub-publication>Nature Biotechnology</span></div><a href=/page/publication/ilastik/><div class=img-hover-zoom><img src=/page/publication/ilastik/featured_hud74f7dc7323e48e35f7fbbb350c4a210_653690_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="Ilastik: interactive machine learning for (bio) image analysis" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/ilastik/>Ilastik: interactive machine learning for (bio) image analysis</a></div><a href=/page/publication/ilastik/ class=summary-link><div class=article-style><p>We present ilastik, an easy-to-use interactive tool that brings machine-learning-based (bio)image analysis to end users without substantial computational expertise. It contains pre-defined workflows for image segmentation, object classification, counting and tracking. Users adapt the workflows to the problem at hand by interactively providing sparse training annotations for a nonlinear classifier. ilastik can process data in up to five dimensions (3D, time and number of channels). Its computational back end runs operations on-demand wherever possible, allowing for interactive prediction on data larger than RAM. Once the classifiers are trained, ilastik workflows can be applied to new data from the command line without further user interaction. We describe all ilastik workflows in detail, including three case studies and a discussion on the expected performance.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.nature.com/articles/s41587-019-0140-0.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/ilastik/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/kipoi/kipoi target=_blank rel=noopener>Code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://kipoi.org/ target=_blank rel=noopener>Project</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Ziga Avsec</span>, <span>Roman Kreuzhuber</span>, <span>Johnny Israeli</span>, <span>Nancy Xu</span>, <span>Jun Cheng</span>, <span>Avanti Shrikumar</span>, <span>Abhimanyu Banerjee</span>, <span>Daniel S. Kim</span>, <span>amdin</span>, <span>Lara Urban</span>, <span>Anshul Kundaje</span>, <span>Oliver Stegle</span>, <span>Julien Gagneur</span></div><span class=article-date>July, 2019</span>
<span class=middot-divider></span>
<span class=pub-publication>Nature Biotechnology</span></div><a href=/page/publication/kipoi/><div class=img-hover-zoom><img src=/page/publication/kipoi/featured_hu90232cb887d0e6448a3aea7617596971_374731_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="The Kipoi repository accelerates community exchange and reuse of predictive models for genomics" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/kipoi/>The Kipoi repository accelerates community exchange and reuse of predictive models for genomics</a></div><a href=/page/publication/kipoi/ class=summary-link><div class=article-style><p>Advances in machine learning, coupled with rapidly growing genome sequencing and molecular profiling datasets, are catalyzing progress in genomics1. In particular, predictive machine learning models, which are mathematical functions trained to map input data to output values, have found widespread usage. Prominent examples include calling variants from whole-genome sequencing data2,3, estimating CRISPR guide activity4,5 and predicting molecular phenotypes, including transcription factor binding, chromatin accessibility and splicing efficiency, from DNA sequence1,6,7,8,9,10,11. Once trained, these models can be probed in silico to infer quantitative relationships between diverse genomic data modalities, enabling several key applications such as the interpretation of functional genetic variants and rational design of synthetic genes.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.nature.com/articles/s41587-019-0140-0.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/kipoi/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/kipoi/kipoi target=_blank rel=noopener>Code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://kipoi.org/ target=_blank rel=noopener>Project</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Constantin Pape</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Peter Li</span>, <span>Viren Jain</span>, <span>Davi D. Bock</span>, <span>Anna Kreshuk</span></div><span class=article-date>June, 2017</span>
<span class=middot-divider></span>
<span class=pub-publication>IEEE Transactions on Medical Imaging</span></div><a href=/page/publication/neuron_segmentation_with/><div class=img-hover-zoom><img src=/page/publication/neuron_segmentation_with/featured_huf135e786a65b25f159a5a4d3e7602041_940903_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="Neuron segmentation with high-level biological priors" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/neuron_segmentation_with/>Neuron segmentation with high-level biological priors</a></div><a href=/page/publication/neuron_segmentation_with/ class=summary-link><div class=article-style><p>We present a novel approach to the problem of neuron segmentation in image volumes acquired by an electron microscopy. Existing methods, such as agglomerative or correlation clustering, rely solely on boundary evidence and have problems where such an evidence is lacking (e.g., incomplete staining) or ambiguous (e.g., co-located cell and mitochondria membranes). We investigate if these difficulties can be overcome by means of sparse region appearance cues that differentiate between pre- and postsynaptic neuron segments in mammalian neural tissue. We combine these cues with the traditional boundary evidence in the asymmetric multiway cut (AMWC) model, which simultaneously solves the partitioning and the semantic region labeling problems. We show that AMWC problems over superpixel graphs can be solved to global optimality with a cutting plane approach, and that the introduction of semantic class priors leads to significantly better segmentations.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://ieeexplore.ieee.org/document/7940079 target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/neuron_segmentation_with/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Constantin Pape</span>, <span>Nasim Rahaman</span>, <span>Timo Prange</span>, <span>Stuart Berg</span>, <span>Davi D. Bock</span>, <span>Albert Cardona</span>, <span>Graham W Knott</span>, <span>Stephen M Plaza</span>, <span>Louis K Scheffer</span>, <span>Ullrich Koethe</span>, <span>Anna Kreshuk</span>, <span>Fred A Hamprecht</span></div><span class=article-date>February, 2017</span>
<span class=middot-divider></span>
<span class=pub-publication>In ICCV 2017</span></div><a href=/page/publication/multicut_brings_automated_segmentation/><div class=img-hover-zoom><img src=/page/publication/multicut_brings_automated_segmentation/featured_huc51ab95e769c5972dedb13894862ea29_16480_808x455_fill_q75_h2_lanczos_smart1.webp height=455 width=808 class=article-banner alt="Multicut brings automated neurite segmentation closer to human performance" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/multicut_brings_automated_segmentation/>Multicut brings automated neurite segmentation closer to human performance</a></div><a href=/page/publication/multicut_brings_automated_segmentation/ class=summary-link><div class=article-style><p>The connectomics community is acquiring volumetric electron microscopy (EM) images of the brain at an unprecedented rate with the aim of mapping out and understanding in detail the physical correlates of information processing in animals. Reliable automatic segmentation is urgently needed for upcoming wholebrain data sets (>100 terabytes (TB) per volume). Manual analysis, despite impressive progress in collaborative annotation1, will not scale to this massive task. We present an algorithm and software package to segment such data sets with low error rates. The software is made available open source in the Supplementary Software and at online repositories, and we also provide precompil</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/217205318/beier_17_multicut.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/multicut_brings_automated_segmentation/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Constantin Pape</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Peter Li</span>, <span>Viren Jain</span>, <span>Davi D. Bock</span>, <span>Anna Kreshuk</span></div><span class=article-date>January, 2017</span>
<span class=middot-divider></span>
<span class=pub-publication>In ICCV 2017</span></div><a href=/page/publication/solving_large_scale/><div class=img-hover-zoom><img src=/page/publication/solving_large_scale/featured_hu6dfc478e6953a31c1dcef9e378b4f83a_11801_808x455_fill_q75_h2_lanczos_smart1.webp height=455 width=808 class=article-banner alt="Solving large multicut problems for connectomics via domain decomposition" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/solving_large_scale/>Solving large multicut problems for connectomics via domain decomposition</a></div><a href=/page/publication/solving_large_scale/ class=summary-link><div class=article-style><p>For image segmentation, recent advances in optimization make it possible to combine noisy region appearance terms with pairwise terms which can not only discourage, but also encourage label transitions, depending on boundary evidence. These models have the potential to overcome problems such as the shrinking bias. However, with the ability to encourage label transitions comes a different problem: strong boundary evidence can overrule weak region appearance terms to create new regions out of nowhere. While some label classes exhibit strong internal boundaries, such as the background class which is the pool of objects. Other label classes, meanwhile, should be modeled as a single region, even if some internal boundaries are visible. We therefore propose in this work to treat label classes asymmetrically: for some classes, we allow a further partitioning into their constituent objects as supported by boundary evidence; for other classes, further partitioning is forbidden. In our experiments, we show where such a model can be useful for both 2D and 3D segmentation.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=http://openaccess.thecvf.com/content_ICCV_2017_workshops/papers/w1/Pape_Solving_Large_Multicut_ICCV_2017_paper.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/solving_large_scale/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Thorben Kroeger</span>, <span>Bjoern Andres</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></div><span class=article-date>October, 2016</span>
<span class=middot-divider></span>
<span class=pub-publication>In ECCV</span></div><a href=/page/publication/efficient_fusion_move/><div class=img-hover-zoom><img src=/page/publication/efficient_fusion_move/featured_hua0ed726f5a5ead4866d7b87ef6f586a9_10248_808x455_fill_q75_h2_lanczos_smart1.webp height=455 width=808 class=article-banner alt="An efficient fusion move algorithm for the minimum cost lifted multicut problem" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/efficient_fusion_move/>An efficient fusion move algorithm for the minimum cost lifted multicut problem</a></div><a href=/page/publication/efficient_fusion_move/ class=summary-link><div class=article-style><p>Many computer vision problems can be cast as an optimization problem whose feasible solutions are decompositions of a graph. The minimum cost lifted multicut problem is such an optimization problem. Its objective function can penalize or reward all decompositions for which any given pair of nodes are in distinct components. While this property has many potential applications, such applications are hampered by the fact that the problem is NP-hard. We propose a fusion move algorithm for computing feasible solutions, better and more efficiently than existing algorithms. We demonstrate this and applications to image segmentation, obtaining a new state of the art for a problem in biological image analysis.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/1939997197/beier_16_efficient.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/efficient_fusion_move/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/DerThorsten/nifty target=_blank rel=noopener>Code</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Nikola Krasowski</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Graham W Knott</span>, <span>Ullrich Koethe</span>, <span>Fred A Hamprecht</span>, <span>Anna Kreshuk</span></div><span class=article-date>April, 2015</span>
<span class=middot-divider></span>
<span class=pub-publication>In ICCV</span></div><a href=/page/publication/improving_3d/><div class=img-hover-zoom><img src=/page/publication/improving_3d/featured_hu2bb203723dac380ae26e75565c1e426e_13419_808x455_fill_q75_h2_lanczos_smart1.webp height=455 width=808 class=article-banner alt="Improving 3D EM data segmentation by joint optimization over boundary evidence and biological priors" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/improving_3d/>Improving 3D EM data segmentation by joint optimization over boundary evidence and biological priors</a></div><a href=/page/publication/improving_3d/ class=summary-link><div class=article-style><p>Beschreibung We present a new automated neuron segmentation algorithm for isotropic 3D electron microscopy data. We cast the problem into the asymmetric multiway cut framework. The latter combines boundary-based segmentation (clustering) with region-based segmentation (semantic labeling) in a single problem and objective function. This joint formulation allows us to augment local boundary evidence with higherlevel biological priors, such as membership to an axonic or dendritic neurite. Joint optimization enforces consistency between evidence and priors, leading to correct resolution of many difficult boundary configurations. We show experimentally on a FIB/SEM dataset of mouse cortex that the new approach outperforms existing hierarchical segmentation and multicut algorithms which only use boundary evidence.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/768610281/krasowski_15_improving.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/improving_3d/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Thorben Kroeger</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></div><span class=article-date>April, 2015</span>
<span class=middot-divider></span>
<span class=pub-publication>In ICCV</span></div><a href=/page/publication/asymetric_cuts/><div class=img-hover-zoom><img src=/page/publication/asymetric_cuts/featured_hu470438db5a2c59940d94b73149c9ca2e_78928_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="Asymmetric Cuts: Joint Image Labeling and Partitioning" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/asymetric_cuts/>Asymmetric Cuts: Joint Image Labeling and Partitioning</a></div><a href=/page/publication/asymetric_cuts/ class=summary-link><div class=article-style><p>For image segmentation, recent advances in optimization make it possible to combine noisy region appearance terms with pairwise terms which can not only discourage, but also encourage label transitions, depending on boundary evidence. These models have the potential to overcome problems such as the shrinking bias. However, with the ability to encourage label transitions comes a different problem: strong boundary evidence can overrule weak region appearance terms to create new regions out of nowhere. While some label classes exhibit strong internal boundaries, such as the background class which is the pool of objects. Other label classes, meanwhile, should be modeled as a single region, even if some internal boundaries are visible. We therefore propose in this work to treat label classes asymmetrically: for some classes, we allow a further partitioning into their constituent objects as supported by boundary evidence; for other classes, further partitioning is forbidden. In our experiments, we show where such a model can be useful for both 2D and 3D segmentation.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/1204284840/kroeger_14_asymmetric.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/asymetric_cuts/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Thorben Kroeger</span>, <span>Fred A. Hamprecht</span>, <span>Jorg H. Kappes</span></div><span class=article-date>January, 2015</span>
<span class=middot-divider></span>
<span class=pub-publication>In ICCV</span></div><a href=/page/publication/fusion_moves_for_correlation_clustering/><div class=img-hover-zoom><img src=/page/publication/fusion_moves_for_correlation_clustering/featured_hu5dc3dd714ff029a113a669dd408019db_6412_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="Fusion moves for correlation clustering" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/fusion_moves_for_correlation_clustering/>Fusion moves for correlation clustering</a></div><a href=/page/publication/fusion_moves_for_correlation_clustering/ class=summary-link><div class=article-style><p>Correlation clustering, or multicut partitioning, is widely used in image segmentation for partitioning an undirected graph or image with positive and negative edge weights such that the sum of cut edge weights is minimized. Due to its NP-hardness, exact solvers do not scale and approximative solvers often give unsatisfactory results. We investigate scalable methods for correlation clustering. To this end we define fusion moves for the correlation clustering problem. Our algorithm iteratively fuses the current and a proposed partitioning which monotonously improves the partitioning and maintains a valid partitioning at all times. Furthermore, it scales to larger datasets, gives near optimal solutions, and at the same time shows a good anytime performance.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://openaccess.thecvf.com/content_cvpr_2015/papers/Beier_Fusion_Moves_for_2015_CVPR_paper.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/fusion_moves_for_correlation_clustering/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Julian Yarkony</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Pierre Baldi</span>, <span>Fred A Hamprecht</span></div><span class=article-date>September, 2014</span>
<span class=middot-divider></span>
<span class=pub-publication>International Workshop on New Frontiers in Mining Complex Patterns</span></div><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/parallel_multicut/>Parallel Multicut Segmentation via Dual Decomposition</a></div><a href=/page/publication/parallel_multicut/ class=summary-link><div class=article-style><p>We propose a new outer relaxation of the multicut polytope, along with a dual decomposition approach for correlation clustering and multicut segmentation, for general graphs. Each subproblem is a minimum -cut problem and can thus be solved efficiently. An optimal reparameterization is found using subgradients and affords a new characterization of the basic LP relaxation of the multicut problem, as well as informed decoding heuristics. The algorithm we propose for solving the problem distributes the computation and is amenable to a parallel implementation.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href="https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.720.7046&rep=rep1&type=pdf" target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/parallel_multicut/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Jörg Hendrik Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Christoph Schnörr</span></div><span class=article-date>September, 2014</span>
<span class=middot-divider></span>
<span class=pub-publication>In ECCV</span></div><a href=/page/publication/map_inference/><div class=img-hover-zoom><img src=/page/publication/map_inference/featured_hu6f28b156f5d116a2429128681859df35_19818_808x455_fill_q75_h2_lanczos_smart1.webp height=455 width=808 class=article-banner alt="MAP-inference on large scale higher-order discrete graphical models by fusion moves" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/map_inference/>MAP-inference on large scale higher-order discrete graphical models by fusion moves</a></div><a href=/page/publication/map_inference/ class=summary-link><div class=article-style><p>Many computer vision problems can be cast into optimization problems over discrete graphical models also known as Markov or conditional random fields. Standard methods are able to solve those problems quite efficiently. However, problems with huge label spaces and or higher-order structure remain challenging or intractable even for approximate methods. We reconsider the work of Lempitsky et al. 2010 on fusion moves and apply it to general discrete graphical models. We propose two alternatives for calculating fusion moves that outperform the standard in several applications. Our generic software framework allows us to easily use different proposal generators which spans a large class of inference algorithms and thus makes exhaustive evaluation feasible. Because these fusion algorithms can be applied to models with huge label spaces and higher-order terms, they might stimulate and support research of such models which may have not been possible so far due to the lack of adequate inference methods.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://link.springer.com/content/pdf/10.1007/978-3-319-16181-5_37.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/map_inference/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/opengm/opengm target=_blank rel=noopener>Code</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Thorben Kroeger</span>, <span>Jorg H. Kappes</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></div><span class=article-date>January, 2014</span>
<span class=middot-divider></span>
<span class=pub-publication>In ICCV</span></div><a href=/page/publication/cut_glue_and_cut/><div class=img-hover-zoom><img src=/page/publication/cut_glue_and_cut/featured_hu7cb38bb6e06a8ec2e961759c0ec132cf_593123_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="Cut, glue & cut: A fast, approximate solver for multicut partitioning" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/cut_glue_and_cut/>Cut, glue & cut: A fast, approximate solver for multicut partitioning</a></div><a href=/page/publication/cut_glue_and_cut/ class=summary-link><div class=article-style><p>Recently, unsupervised image segmentation has become increasingly popular. Starting from a superpixel segmentation, an edge-weighted region adjacency graph is constructed. Amongst all segmentations of the graph, the one which best conforms to the given image evidence, as measured by the sum of cut edge weights, is chosen. Since this problem is NP-hard, we propose a new approximate solver based on the move-making paradigm: first, the graph is recursively partitioned into small regions (cut phase). Then, for any two adjacent regions, we consider alternative cuts of these two regions defining possible moves (glue & cut phase). For planar problems, the optimal move can be found, whereas for non-planar problems, efficient approximations exist. We evaluate our algorithm on published and new benchmark datasets, which we make available here. The proposed algorithm finds segmentations that, as measured by a loss function, are as close to the ground-truth as the global optimum found by exact solvers. It does so significantly faster then existing approximate methods, which is important for large-scale problems.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.cv-foundation.org/openaccess/content_cvpr_2014/papers/Beier_Cut_Glue__2014_CVPR_paper.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/cut_glue_and_cut/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Bjoern Andres</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></div><span class=article-date>November, 2012</span>
<span class=middot-divider></span>
<span class=pub-publication>In ECCV</span></div><a href=/page/publication/lazy_flipper/><div class=img-hover-zoom><img src=/page/publication/lazy_flipper/featured_huf99e1ff45eafab1956c20888fc623fb8_45813_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="The Lazy Flipper: Efficient Depth-Limited Exhaustive Search in Discrete Graphical Models" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/lazy_flipper/>The Lazy Flipper: Efficient Depth-Limited Exhaustive Search in Discrete Graphical Models</a></div><a href=/page/publication/lazy_flipper/ class=summary-link><div class=article-style><p>We propose a novel graphical model for probabilistic image segmentation that contributes both to aspects of perceptual grouping in connection with image segmentation, and to globally optimal inference with higher-order graphical models.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=http://ipa.iwr.uni-heidelberg.de/dokuwiki/Papers/andres2011probabilistic.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/lazy_flipper/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Bjoern Andres</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Köthe</span>, <span>Fred A Hamprecht</span></div><span class=article-date>October, 2012</span>
<span class=middot-divider></span>
<span class=pub-publication>arXiv preprint arXiv:1206.0111</span></div><a href=/page/publication/opengm/><div class=img-hover-zoom><img src=/page/publication/opengm/featured_hu8b8dc628d345cfb9e150bcc573382a9d_5797_808x455_fill_q75_h2_lanczos_smart1_3.webp height=455 width=808 class=article-banner alt="OpenGM: A C++ library for discrete graphical models" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/opengm/>OpenGM: A C++ library for discrete graphical models</a></div><a href=/page/publication/opengm/ class=summary-link><div class=article-style><p>We propose a new exhaustive search algorithm for optimization in discrete graphical models. When pursued to the full search depth (typically intractable), it is guaranteed to converge to a global optimum, passing through a series of monotonously improving local optima that are guaranteed to be optimal within a given and increasing Hamming distance.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://link.springer.com/content/pdf/10.1007/978-3-642-33786-4_12.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/opengm/cite.bib>Cite</a></div></div><div class="card-simple view-card"><div class=article-metadata><div><span>Bjoern Andres</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></div><span class=article-date>November, 2011</span>
<span class=middot-divider></span>
<span class=pub-publication>In ICCV</span></div><a href=/page/publication/probabilistic_image_segmentation/><div class=img-hover-zoom><img src=/page/publication/probabilistic_image_segmentation/featured_hu1d972d011f8040c6a0c9c48c2912fbba_10450_808x455_fill_q75_h2_lanczos_smart1.webp height=455 width=808 class=article-banner alt="Probabilistic Image Segmentation with Closedness Constraints" loading=lazy></div></a><div class="section-subheading article-title mb-1 mt-3"><a href=/page/publication/probabilistic_image_segmentation/>Probabilistic Image Segmentation with Closedness Constraints</a></div><a href=/page/publication/probabilistic_image_segmentation/ class=summary-link><div class=article-style><p>We propose a novel graphical model for probabilistic image segmentation that contributes both to aspects of perceptual grouping in connection with image segmentation, and to globally optimal inference with higher-order graphical models.</p></div></a><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=http://ipa.iwr.uni-heidelberg.de/dokuwiki/Papers/andres2011probabilistic.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/probabilistic_image_segmentation/cite.bib>Cite</a></div></div></div></div></div></section><section id=publications class="home-section wg-pages"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Recent Publications</h1></div><div class="col-12 col-lg-8"><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Stuart Berg</span>, <span>Dominik Kutra</span>, <span>Thorben Kroeger</span>, <span>Christoph N. Straehle</span>, <span>Bernhard X. Kausler</span>, <span>Carsten Haubold</span>, <span>Martin Schiegg</span>, <span>Janez Ales</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Markus Rudy</span>, <span>Kemal Eren</span>, <span>Jaime I Cervantes</span>, <span>Buote Xu</span>, <span>Fynn Beuttenmueller</span>, <span>Adrian Wolny</span>, <span>Chong Zhang</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span>, <span>Anna Kreshuk</span></span>
(2019).
<a href=/page/publication/ilastik/>Ilastik: interactive machine learning for (bio) image analysis</a>.
Nature Biotechnology.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.nature.com/articles/s41587-019-0140-0.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/ilastik/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/kipoi/kipoi target=_blank rel=noopener>Code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://kipoi.org/ target=_blank rel=noopener>Project</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Ziga Avsec</span>, <span>Roman Kreuzhuber</span>, <span>Johnny Israeli</span>, <span>Nancy Xu</span>, <span>Jun Cheng</span>, <span>Avanti Shrikumar</span>, <span>Abhimanyu Banerjee</span>, <span>Daniel S. Kim</span>, <span>amdin</span>, <span>Lara Urban</span>, <span>Anshul Kundaje</span>, <span>Oliver Stegle</span>, <span>Julien Gagneur</span></span>
(2019).
<a href=/page/publication/kipoi/>The Kipoi repository accelerates community exchange and reuse of predictive models for genomics</a>.
Nature Biotechnology.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.nature.com/articles/s41587-019-0140-0.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/kipoi/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/kipoi/kipoi target=_blank rel=noopener>Code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://kipoi.org/ target=_blank rel=noopener>Project</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Constantin Pape</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Peter Li</span>, <span>Viren Jain</span>, <span>Davi D. Bock</span>, <span>Anna Kreshuk</span></span>
(2017).
<a href=/page/publication/neuron_segmentation_with/>Neuron segmentation with high-level biological priors</a>.
IEEE Transactions on Medical Imaging.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://ieeexplore.ieee.org/document/7940079 target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/neuron_segmentation_with/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Constantin Pape</span>, <span>Nasim Rahaman</span>, <span>Timo Prange</span>, <span>Stuart Berg</span>, <span>Davi D. Bock</span>, <span>Albert Cardona</span>, <span>Graham W Knott</span>, <span>Stephen M Plaza</span>, <span>Louis K Scheffer</span>, <span>Ullrich Koethe</span>, <span>Anna Kreshuk</span>, <span>Fred A Hamprecht</span></span>
(2017).
<a href=/page/publication/multicut_brings_automated_segmentation/>Multicut brings automated neurite segmentation closer to human performance</a>.
In ICCV 2017.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/217205318/beier_17_multicut.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/multicut_brings_automated_segmentation/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Constantin Pape</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Peter Li</span>, <span>Viren Jain</span>, <span>Davi D. Bock</span>, <span>Anna Kreshuk</span></span>
(2017).
<a href=/page/publication/solving_large_scale/>Solving large multicut problems for connectomics via domain decomposition</a>.
In ICCV 2017.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=http://openaccess.thecvf.com/content_ICCV_2017_workshops/papers/w1/Pape_Solving_Large_Multicut_ICCV_2017_paper.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/solving_large_scale/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Thorben Kroeger</span>, <span>Bjoern Andres</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></span>
(2016).
<a href=/page/publication/efficient_fusion_move/>An efficient fusion move algorithm for the minimum cost lifted multicut problem</a>.
In ECCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/1939997197/beier_16_efficient.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/efficient_fusion_move/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/DerThorsten/nifty target=_blank rel=noopener>Code</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Nikola Krasowski</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Graham W Knott</span>, <span>Ullrich Koethe</span>, <span>Fred A Hamprecht</span>, <span>Anna Kreshuk</span></span>
(2015).
<a href=/page/publication/improving_3d/>Improving 3D EM data segmentation by joint optimization over boundary evidence and biological priors</a>.
In ICCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/768610281/krasowski_15_improving.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/improving_3d/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Thorben Kroeger</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></span>
(2015).
<a href=/page/publication/asymetric_cuts/>Asymmetric Cuts: Joint Image Labeling and Partitioning</a>.
In ICCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/1204284840/kroeger_14_asymmetric.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/asymetric_cuts/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Thorben Kroeger</span>, <span>Fred A. Hamprecht</span>, <span>Jorg H. Kappes</span></span>
(2015).
<a href=/page/publication/fusion_moves_for_correlation_clustering/>Fusion moves for correlation clustering</a>.
In ICCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://openaccess.thecvf.com/content_cvpr_2015/papers/Beier_Fusion_Moves_for_2015_CVPR_paper.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/fusion_moves_for_correlation_clustering/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Julian Yarkony</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Pierre Baldi</span>, <span>Fred A Hamprecht</span></span>
(2014).
<a href=/page/publication/parallel_multicut/>Parallel Multicut Segmentation via Dual Decomposition</a>.
International Workshop on New Frontiers in Mining Complex Patterns.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href="https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.720.7046&rep=rep1&type=pdf" target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/parallel_multicut/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Jörg Hendrik Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Christoph Schnörr</span></span>
(2014).
<a href=/page/publication/map_inference/>MAP-inference on large scale higher-order discrete graphical models by fusion moves</a>.
In ECCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://link.springer.com/content/pdf/10.1007/978-3-319-16181-5_37.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/map_inference/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/opengm/opengm target=_blank rel=noopener>Code</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Thorben Kroeger</span>, <span>Jorg H. Kappes</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></span>
(2014).
<a href=/page/publication/cut_glue_and_cut/>Cut, glue & cut: A fast, approximate solver for multicut partitioning</a>.
In ICCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.cv-foundation.org/openaccess/content_cvpr_2014/papers/Beier_Cut_Glue__2014_CVPR_paper.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/cut_glue_and_cut/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Bjoern Andres</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></span>
(2012).
<a href=/page/publication/lazy_flipper/>The Lazy Flipper: Efficient Depth-Limited Exhaustive Search in Discrete Graphical Models</a>.
In ECCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=http://ipa.iwr.uni-heidelberg.de/dokuwiki/Papers/andres2011probabilistic.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/lazy_flipper/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Bjoern Andres</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Köthe</span>, <span>Fred A Hamprecht</span></span>
(2012).
<a href=/page/publication/opengm/>OpenGM: A C++ library for discrete graphical models</a>.
arXiv preprint arXiv:1206.0111.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://link.springer.com/content/pdf/10.1007/978-3-642-33786-4_12.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/opengm/cite.bib>Cite</a></p></div><div class="pub-list-item view-citation" style=margin-bottom:1rem><i class="far fa-file-alt pub-icon" aria-hidden=true></i>
<span class="article-metadata li-cite-author"><span>Bjoern Andres</span>, <span>Jorg H. Kappes</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span></span>
(2011).
<a href=/page/publication/probabilistic_image_segmentation/>Probabilistic Image Segmentation with Closedness Constraints</a>.
In ICCV.<p><a class="btn btn-outline-primary btn-page-header btn-sm" href=http://ipa.iwr.uni-heidelberg.de/dokuwiki/Papers/andres2011probabilistic.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/probabilistic_image_segmentation/cite.bib>Cite</a></p></div></div></div></div></section><section id=tags class="home-section wg-tag-cloud"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Popular Topics</h1></div><div class="col-12 col-lg-8"><div class=tag-cloud><a href=/page/tag/agglomerative-clustering/ style=font-size:.7rem>Agglomerative Clustering</a>
<a href=/page/tag/box2d/ style=font-size:1.259879525495411rem>Box2D</a>
<a href=/page/tag/c-/ style=font-size:.7rem>C++</a>
<a href=/page/tag/conda/ style=font-size:.7rem>Conda</a>
<a href=/page/tag/conda-forge/ style=font-size:.7rem>conda-forge</a>
<a href=/page/tag/deep-learning/ style=font-size:.7rem>Deep Learning</a>
<a href=/page/tag/discrete-optimization/ style=font-size:1.259879525495411rem>Discrete Optimization</a>
<a href=/page/tag/emscripten-forge/ style=font-size:1.259879525495411rem>Emscripten-Forge</a>
<a href=/page/tag/games/ style=font-size:1.259879525495411rem>Games</a>
<a href=/page/tag/jupyter/ style=font-size:1.259879525495411rem>Jupyter</a>
<a href=/page/tag/jupyterlite/ style=font-size:1.587388052831781rem>JupyterLite</a>
<a href=/page/tag/lifted-multicut/ style=font-size:.7rem>Lifted Multicut</a>
<a href=/page/tag/mamba/ style=font-size:.7rem>Mamba</a>
<a href=/page/tag/multicut/ style=font-size:.7rem>Multicut</a>
<a href=/page/tag/wasm/ style=font-size:1.819759050990822rem>Wasm</a>
<a href=/page/tag/xeus/ style=font-size:1.259879525495411rem>Xeus</a></div></div></div></div></section><section id=contact class="home-section wg-contact"><div class=home-section-bg></div><div class=container><div class=row><div class="section-heading col-12 col-lg-4 mb-3 mb-lg-0 d-flex flex-column align-items-center align-items-lg-start"><h1 class=mb-0>Contact</h1></div><div class="col-12 col-lg-8"><div class=mb-3><form name=contact method=post netlify netlify-honeypot=_gotcha><div class="form-group form-inline"><label class=sr-only for=inputName>Name</label>
<input type=text name=name class="form-control w-100" id=inputName placeholder=Name required></div><div class="form-group form-inline"><label class=sr-only for=inputEmail>Email</label>
<input type=email name=email class="form-control w-100" id=inputEmail placeholder=Email required></div><div class=form-group><label class=sr-only for=inputMessage>Message</label>
<textarea name=message class=form-control id=inputMessage rows=5 placeholder=Message required></textarea></div><div class=d-none><label>Do not fill this field unless you are a bot: <input name=_gotcha></label></div><button type=submit class="btn btn-primary px-3 py-2 w-100">Send</button></form></div><ul class=fa-ul><li><i class="fa-li fas fa-envelope fa-2x" aria-hidden=true></i>
<span id=person-email><a href=mailto:thorsten.beier@quantstack.net>thorsten.beier@quantstack.net</a></span></li><li><i class="fa-li fab fa-twitter fa-2x" aria-hidden=true></i>
<a href=https://twitter.com/ThorstenBeier target=_blank rel=noopener>DM Me</a></li></ul></div></div></div></section></div><div class=page-footer><div class=container><footer class=site-footer><p class="powered-by copyright-license-text">© 2022 Me. This work is licensed under <a href=https://creativecommons.org/licenses/by-nc-nd/4.0 rel="noopener noreferrer" target=_blank>CC BY NC ND 4.0</a></p><p class="powered-by footer-license-icons"><a href=https://creativecommons.org/licenses/by-nc-nd/4.0 rel="noopener noreferrer" target=_blank aria-label="Creative Commons"><i class="fab fa-creative-commons fa-2x" aria-hidden=true></i>
<i class="fab fa-creative-commons-by fa-2x" aria-hidden=true></i>
<i class="fab fa-creative-commons-nc fa-2x" aria-hidden=true></i>
<i class="fab fa-creative-commons-nd fa-2x" aria-hidden=true></i></a></p><p class=powered-by>Published with <a href="https://wowchemy.com/?utm_campaign=poweredby" target=_blank rel=noopener>Wowchemy</a> — the free, <a href=https://github.com/wowchemy/wowchemy-hugo-themes target=_blank rel=noopener>open source</a> website builder that empowers creators.</p></footer></div></div><script src=/page/js/vendor-bundle.min.d26509351aa0ff874abbee824e982e9b.js></script>
<script src=https://cdn.jsdelivr.net/gh/desandro/imagesloaded@v4.1.4/imagesloaded.pkgd.min.js integrity="sha512-S5PZ9GxJZO16tT9r3WJp/Safn31eu8uWrzglMahDT4dsmgqWonRY9grk3j+3tfuPr9WJNsfooOR7Gi7HL5W2jw==" crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/metafizzy/isotope@v3.0.6/dist/isotope.pkgd.min.js integrity="sha512-Zq2BOxyhvnRFXu0+WE6ojpZLOU2jdnqbrM1hmVdGzyeCa1DgM3X5Q4A/Is9xA1IkbUeDd7755dNNI/PzSf2Pew==" crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/highlight.min.js integrity="sha512-Ypjm0o7jOxAd4hpdoppSEN0TQOC19UtPAqD+4s5AlXmUvbmmS/YMxYqAqarQYyxTnB6/rqip9qcxlNB/3U9Wdg==" crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/languages/r.min.js crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/languages/latex.min.js crossorigin=anonymous></script>
<script id=search-hit-fuse-template type=text/x-template>
        <div class="search-hit" id="summary-{{key}}">
          <div class="search-hit-content">
            <div class="search-hit-name">
              <a href="{{relpermalink}}">{{title}}</a>
              <div class="article-metadata search-hit-type">{{type}}</div>
              <p class="search-hit-description">{{snippet}}</p>
            </div>
          </div>
        </div>
      </script><script src=https://cdn.jsdelivr.net/gh/krisk/Fuse@v3.2.1/dist/fuse.min.js integrity="sha512-o38bmzBGX+hD3JHWUFCDA09btWaqrNmoJ3RXLlrysA7PP01Kgs4UlE4MhelE1v5dJR3+cxlR4qQlotsW7jKsnw==" crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/julmot/mark.js@8.11.1/dist/jquery.mark.min.js integrity="sha512-mhbv5DqBMgrWL+32MmsDOt/OAvqr/cHimk6B8y/bx/xS88MVkYGPiVv2ixKVrkywF2qHplNRUvFsAHUdxZ3Krg==" crossorigin=anonymous></script>
<script id=page-data type=application/json>{"use_headroom":false}</script><script src=/page/en/js/wowchemy.min.43c8f7a4851160885a7d8069dfa86538.js></script><div id=modal class="modal fade" role=dialog><div class=modal-dialog><div class=modal-content><div class=modal-header><h5 class=modal-title>Cite</h5><button type=button class=close data-dismiss=modal aria-label=Close>
<span aria-hidden=true>&#215;</span></button></div><div class=modal-body><pre><code class="tex hljs"></code></pre></div><div class=modal-footer><a class="btn btn-outline-primary my-1 js-copy-cite" href=# target=_blank><i class="fas fa-copy"></i> Copy</a>
<a class="btn btn-outline-primary my-1 js-download-cite" href=# target=_blank><i class="fas fa-download"></i> Download</a><div id=modal-error></div></div></div></div></div><script src=/page/js/wowchemy-publication.68f8d7090562ca65fc6d3cb3f8f2d2cb.js type=module></script></body></html>