<!doctype html><html lang=en-us><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=generator content="Wowchemy 5.5.0 for Hugo"><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=preload as=style href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap"><link rel=stylesheet href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&family=Roboto+Mono&family=Roboto:wght@400;700&display=swap" media=print onload='this.media="all"'><meta name=author content="Dr. Thorsten Beier"><meta name=description content="A highly-customizable Hugo academic resume theme powered by Wowchemy website builder."><link rel=alternate hreflang=en-us href=https://derthorsten.github.io/page/publication-type/2/><meta name=theme-color content="#1565c0"><link rel=stylesheet href=/page/css/vendor-bundle.min.c7b8d9abd591ba2253ea42747e3ac3f5.css media=print onload='this.media="all"'><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css integrity="sha512-W0xM4mr6dEP9nREo7Z9z+9X70wytKvMGeDsj7ps2+xg5QPrEBXC8tAW1IFnzjR6eoJ90JmCnFzerQJTLzIEHjA==" crossorigin=anonymous media=print onload='this.media="all"'><link rel=stylesheet href=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/styles/github.min.css crossorigin=anonymous title=hl-light media=print onload='this.media="all"'><link rel=stylesheet href=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/styles/dracula.min.css crossorigin=anonymous title=hl-dark media=print onload='this.media="all"' disabled><link rel=stylesheet href=/page/css/wowchemy.1052fab8b7700a3dc49ee23683097d66.css><link rel=alternate href=/page/publication-type/2/index.xml type=application/rss+xml title="Dr. Thorsten Beier"><link rel=manifest href=/page/manifest.webmanifest><link rel=icon type=image/png href=/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_32x32_fill_lanczos_center_3.png><link rel=apple-touch-icon type=image/png href=/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_180x180_fill_lanczos_center_3.png><link rel=canonical href=https://derthorsten.github.io/page/publication-type/2/><meta property="twitter:card" content="summary"><meta property="twitter:site" content="@wowchemy"><meta property="twitter:creator" content="@wowchemy"><meta property="og:site_name" content="Dr. Thorsten Beier"><meta property="og:url" content="https://derthorsten.github.io/page/publication-type/2/"><meta property="og:title" content="2 | Dr. Thorsten Beier"><meta property="og:description" content="A highly-customizable Hugo academic resume theme powered by Wowchemy website builder."><meta property="og:image" content="https://derthorsten.github.io/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_512x512_fill_lanczos_center_3.png"><meta property="twitter:image" content="https://derthorsten.github.io/page/media/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_512x512_fill_lanczos_center_3.png"><meta property="og:locale" content="en-us"><meta property="og:updated_time" content="2019-12-01T00:00:00+00:00"><title>2 | Dr. Thorsten Beier</title></head><body id=top data-spy=scroll data-offset=70 data-target=#TableOfContents class=page-wrapper><script src=/page/js/wowchemy-init.min.2ed908358299dd7ab553faae685c746c.js></script><aside class=search-modal id=search><div class=container><section class=search-header><div class="row no-gutters justify-content-between mb-3"><div class=col-6><h1>Search</h1></div><div class="col-6 col-search-close"><a class=js-search href=# aria-label=Close><i class="fas fa-times-circle text-muted" aria-hidden=true></i></a></div></div><div id=search-box><input name=q id=search-query placeholder=Search... autocapitalize=off autocomplete=off autocorrect=off spellcheck=false type=search class=form-control aria-label=Search...></div></section><section class=section-search-results><div id=search-hits></div></section></div></aside><div class=page-header><header class=header--fixed><nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main><div class=container-xl><div class="d-none d-lg-inline-flex"><a class=navbar-brand href=/>Dr. Thorsten Beier</a></div><button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar-content aria-expanded=false aria-label="Toggle navigation">
<span><i class="fas fa-bars"></i></span></button><div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none"><a class=navbar-brand href=/>Dr. Thorsten Beier</a></div><div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content><ul class="navbar-nav d-md-inline-flex"><li class=nav-item><a class=nav-link href=/page/#about><span>Home</span></a></li><li class=nav-item><a class=nav-link href=/page/#posts><span>Posts</span></a></li><li class=nav-item><a class=nav-link href=/page/#projects><span>Projects</span></a></li><li class=nav-item><a class=nav-link href=/page/#talks><span>Talks</span></a></li><li class=nav-item><a class=nav-link href=/page/#featured><span>Publications</span></a></li><li class=nav-item><a class=nav-link href=/page/#contact><span>Contact</span></a></li></ul></div><ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2"><li class=nav-item><a class="nav-link js-search" href=# aria-label=Search><i class="fas fa-search" aria-hidden=true></i></a></li><li class="nav-item dropdown theme-dropdown"><a href=# class=nav-link data-toggle=dropdown aria-haspopup=true aria-label="Display preferences"><i class="fas fa-moon" aria-hidden=true></i></a><div class=dropdown-menu><a href=# class="dropdown-item js-set-theme-light"><span>Light</span></a>
<a href=# class="dropdown-item js-set-theme-dark"><span>Dark</span></a>
<a href=# class="dropdown-item js-set-theme-auto"><span>Automatic</span></a></div></li></ul></div></nav></header></div><div class=page-body><div class="universal-wrapper pt-3"><h1>2</h1></div><div class=universal-wrapper><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/publication/ilastik/>Ilastik: interactive machine learning for (bio) image analysis</a></div><a href=/page/publication/ilastik/ class=summary-link><div class=article-style>We present ilastik, an easy-to-use interactive tool that brings machine-learning-based (bio)image analysis to end users without substantial computational expertise. It contains pre-defined workflows for image segmentation, object classification, counting and tracking. Users adapt the workflows to the problem at hand by interactively providing sparse training annotations for a nonlinear classifier. ilastik can process data in up to five dimensions (3D, time and number of channels). Its computational back end runs operations on-demand wherever possible, allowing for interactive prediction on data larger than RAM. Once the classifiers are trained, ilastik workflows can be applied to new data from the command line without further user interaction. We describe all ilastik workflows in detail, including three case studies and a discussion on the expected performance.</div></a><div class="stream-meta article-metadata"><div><span>Stuart Berg</span>, <span>Dominik Kutra</span>, <span>Thorben Kroeger</span>, <span>Christoph N. Straehle</span>, <span>Bernhard X. Kausler</span>, <span>Carsten Haubold</span>, <span>Martin Schiegg</span>, <span>Janez Ales</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Markus Rudy</span>, <span>Kemal Eren</span>, <span>Jaime I Cervantes</span>, <span>Buote Xu</span>, <span>Fynn Beuttenmueller</span>, <span>Adrian Wolny</span>, <span>Chong Zhang</span>, <span>Ullrich Koethe</span>, <span>Fred A. Hamprecht</span>, <span>Anna Kreshuk</span></div></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.nature.com/articles/s41587-019-0140-0.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/ilastik/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/kipoi/kipoi target=_blank rel=noopener>Code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://kipoi.org/ target=_blank rel=noopener>Project</a></div></div><div class=ml-3><a href=/page/publication/ilastik/><img src=/page/publication/ilastik/featured_hud74f7dc7323e48e35f7fbbb350c4a210_653690_150x0_resize_q75_h2_lanczos_3.webp height=82 width=150 alt="Ilastik: interactive machine learning for (bio) image analysis" loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/publication/kipoi/>The Kipoi repository accelerates community exchange and reuse of predictive models for genomics</a></div><a href=/page/publication/kipoi/ class=summary-link><div class=article-style>Advances in machine learning, coupled with rapidly growing genome sequencing and molecular profiling datasets, are catalyzing progress in genomics1. In particular, predictive machine learning models, which are mathematical functions trained to map input data to output values, have found widespread usage. Prominent examples include calling variants from whole-genome sequencing data2,3, estimating CRISPR guide activity4,5 and predicting molecular phenotypes, including transcription factor binding, chromatin accessibility and splicing efficiency, from DNA sequence1,6,7,8,9,10,11. Once trained, these models can be probed in silico to infer quantitative relationships between diverse genomic data modalities, enabling several key applications such as the interpretation of functional genetic variants and rational design of synthetic genes.</div></a><div class="stream-meta article-metadata"><div><span>Ziga Avsec</span>, <span>Roman Kreuzhuber</span>, <span>Johnny Israeli</span>, <span>Nancy Xu</span>, <span>Jun Cheng</span>, <span>Avanti Shrikumar</span>, <span>Abhimanyu Banerjee</span>, <span>Daniel S. Kim</span>, <span>amdin</span>, <span>Lara Urban</span>, <span>Anshul Kundaje</span>, <span>Oliver Stegle</span>, <span>Julien Gagneur</span></div></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://www.nature.com/articles/s41587-019-0140-0.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/kipoi/cite.bib>Cite</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://github.com/kipoi/kipoi target=_blank rel=noopener>Code</a>
<a class="btn btn-outline-primary btn-page-header btn-sm" href=https://kipoi.org/ target=_blank rel=noopener>Project</a></div></div><div class=ml-3><a href=/page/publication/kipoi/><img src=/page/publication/kipoi/featured_hu90232cb887d0e6448a3aea7617596971_374731_150x0_resize_q75_h2_lanczos_3.webp height=70 width=150 alt="The Kipoi repository accelerates community exchange and reuse of predictive models for genomics" loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/publication/neuron_segmentation_with/>Neuron segmentation with high-level biological priors</a></div><a href=/page/publication/neuron_segmentation_with/ class=summary-link><div class=article-style>We present a novel approach to the problem of neuron segmentation in image volumes acquired by an electron microscopy. Existing methods, such as agglomerative or correlation clustering, rely solely on boundary evidence and have problems where such an evidence is lacking (e.g., incomplete staining) or ambiguous (e.g., co-located cell and mitochondria membranes). We investigate if these difficulties can be overcome by means of sparse region appearance cues that differentiate between pre- and postsynaptic neuron segments in mammalian neural tissue. We combine these cues with the traditional boundary evidence in the asymmetric multiway cut (AMWC) model, which simultaneously solves the partitioning and the semantic region labeling problems. We show that AMWC problems over superpixel graphs can be solved to global optimality with a cutting plane approach, and that the introduction of semantic class priors leads to significantly better segmentations.</div></a><div class="stream-meta article-metadata"><div><span>Constantin Pape</span>, <span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Peter Li</span>, <span>Viren Jain</span>, <span>Davi D. Bock</span>, <span>Anna Kreshuk</span></div></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://ieeexplore.ieee.org/document/7940079 target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/neuron_segmentation_with/cite.bib>Cite</a></div></div><div class=ml-3><a href=/page/publication/neuron_segmentation_with/><img src=/page/publication/neuron_segmentation_with/featured_huf135e786a65b25f159a5a4d3e7602041_940903_150x0_resize_q75_h2_lanczos_3.webp height=119 width=150 alt="Neuron segmentation with high-level biological priors" loading=lazy></a></div></div><div class="media stream-item view-compact"><div class=media-body><div class="section-subheading article-title mb-0 mt-0"><a href=/page/publication/multicut_brings_automated_segmentation/>Multicut brings automated neurite segmentation closer to human performance</a></div><a href=/page/publication/multicut_brings_automated_segmentation/ class=summary-link><div class=article-style>The connectomics community is acquiring volumetric electron microscopy (EM) images of the brain at an unprecedented rate with the aim of mapping out and understanding in detail the physical correlates of information processing in animals. Reliable automatic segmentation is urgently needed for upcoming wholebrain data sets (>100 terabytes (TB) per volume). Manual analysis, despite impressive progress in collaborative annotation1, will not scale to this massive task. We present an algorithm and software package to segment such data sets with low error rates. The software is made available open source in the Supplementary Software and at online repositories, and we also provide precompil</div></a><div class="stream-meta article-metadata"><div><span class=author-highlighted>Dr. Thorsten Beier</span>, <span>Constantin Pape</span>, <span>Nasim Rahaman</span>, <span>Timo Prange</span>, <span>Stuart Berg</span>, <span>Davi D. Bock</span>, <span>Albert Cardona</span>, <span>Graham W Knott</span>, <span>Stephen M Plaza</span>, <span>Louis K Scheffer</span>, <span>Ullrich Koethe</span>, <span>Anna Kreshuk</span>, <span>Fred A Hamprecht</span></div></div><div class=btn-links><a class="btn btn-outline-primary btn-page-header btn-sm" href=https://hci.iwr.uni-heidelberg.de/sites/default/files/publications/files/217205318/beier_17_multicut.pdf target=_blank rel=noopener>PDF</a>
<a href=# class="btn btn-outline-primary btn-page-header btn-sm js-cite-modal" data-filename=/page/publication/multicut_brings_automated_segmentation/cite.bib>Cite</a></div></div><div class=ml-3><a href=/page/publication/multicut_brings_automated_segmentation/><img src=/page/publication/multicut_brings_automated_segmentation/featured_huc51ab95e769c5972dedb13894862ea29_16480_150x0_resize_q75_h2_lanczos.webp height=150 width=150 alt="Multicut brings automated neurite segmentation closer to human performance" loading=lazy></a></div></div></div></div><div class=page-footer><div class=container><footer class=site-footer><p class="powered-by copyright-license-text">© 2022 Me. This work is licensed under <a href=https://creativecommons.org/licenses/by-nc-nd/4.0 rel="noopener noreferrer" target=_blank>CC BY NC ND 4.0</a></p><p class="powered-by footer-license-icons"><a href=https://creativecommons.org/licenses/by-nc-nd/4.0 rel="noopener noreferrer" target=_blank aria-label="Creative Commons"><i class="fab fa-creative-commons fa-2x" aria-hidden=true></i>
<i class="fab fa-creative-commons-by fa-2x" aria-hidden=true></i>
<i class="fab fa-creative-commons-nc fa-2x" aria-hidden=true></i>
<i class="fab fa-creative-commons-nd fa-2x" aria-hidden=true></i></a></p><p class=powered-by>Published with <a href="https://wowchemy.com/?utm_campaign=poweredby" target=_blank rel=noopener>Wowchemy</a> — the free, <a href=https://github.com/wowchemy/wowchemy-hugo-themes target=_blank rel=noopener>open source</a> website builder that empowers creators.</p></footer></div></div><script src=/page/js/vendor-bundle.min.d26509351aa0ff874abbee824e982e9b.js></script>
<script src=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/highlight.min.js integrity="sha512-Ypjm0o7jOxAd4hpdoppSEN0TQOC19UtPAqD+4s5AlXmUvbmmS/YMxYqAqarQYyxTnB6/rqip9qcxlNB/3U9Wdg==" crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/languages/r.min.js crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.2.1/build/languages/latex.min.js crossorigin=anonymous></script>
<script id=search-hit-fuse-template type=text/x-template>
        <div class="search-hit" id="summary-{{key}}">
          <div class="search-hit-content">
            <div class="search-hit-name">
              <a href="{{relpermalink}}">{{title}}</a>
              <div class="article-metadata search-hit-type">{{type}}</div>
              <p class="search-hit-description">{{snippet}}</p>
            </div>
          </div>
        </div>
      </script><script src=https://cdn.jsdelivr.net/gh/krisk/Fuse@v3.2.1/dist/fuse.min.js integrity="sha512-o38bmzBGX+hD3JHWUFCDA09btWaqrNmoJ3RXLlrysA7PP01Kgs4UlE4MhelE1v5dJR3+cxlR4qQlotsW7jKsnw==" crossorigin=anonymous></script>
<script src=https://cdn.jsdelivr.net/gh/julmot/mark.js@8.11.1/dist/jquery.mark.min.js integrity="sha512-mhbv5DqBMgrWL+32MmsDOt/OAvqr/cHimk6B8y/bx/xS88MVkYGPiVv2ixKVrkywF2qHplNRUvFsAHUdxZ3Krg==" crossorigin=anonymous></script>
<script id=page-data type=application/json>{"use_headroom":true}</script><script src=/page/js/wowchemy-headroom.c251366b4128fd5e6b046d4c97a62a51.js type=module></script>
<script src=/page/en/js/wowchemy.min.43c8f7a4851160885a7d8069dfa86538.js></script><div id=modal class="modal fade" role=dialog><div class=modal-dialog><div class=modal-content><div class=modal-header><h5 class=modal-title>Cite</h5><button type=button class=close data-dismiss=modal aria-label=Close>
<span aria-hidden=true>&#215;</span></button></div><div class=modal-body><pre><code class="tex hljs"></code></pre></div><div class=modal-footer><a class="btn btn-outline-primary my-1 js-copy-cite" href=# target=_blank><i class="fas fa-copy"></i> Copy</a>
<a class="btn btn-outline-primary my-1 js-download-cite" href=# target=_blank><i class="fas fa-download"></i> Download</a><div id=modal-error></div></div></div></div></div><script src=/page/js/wowchemy-publication.68f8d7090562ca65fc6d3cb3f8f2d2cb.js type=module></script></body></html>